### PR TITLE
[WIP] Add proper window support

### PR DESCRIPTION
**Pull request by MrRio**
Adds window switching and title support to the main menu, and the dock menu.

zukzs1tpoc

screen shot 2016-08-02 at 02 51 56

Adds:
* Switching the window title when the tabs focus changes
* Add the window titles to the Window menu
* Add the titles to the dock menu
* Allow making a new window from the dock

### CODE REVIEW

File: app/menu.js [hunk 1]
```diff
@@ -209,6 +209,7 @@ module.exports = function createMenu ({ createWindow, updatePlugins }) {
     },
     {
       label: 'Window',
```
Comment (sindresorhus) [inline app/menu.js]: `label` is not needed when specifying `role`.

File: app/index.js [hunk 2]
```diff
@@ -309,6 +310,16 @@ app.on('ready', () => {
       }
     }));
 
+    // If we're on Mac make a Dock Menu
+    if (process.platform === 'darwin') {
+      const {app, Menu} = require('electron');
+
```
Comment (rauchg) [inline app/index.js]: extra `\n`

File: app/index.js [hunk 3]
```diff
@@ -309,6 +310,16 @@ app.on('ready', () => {
       }
     }));
 
+    // If we're on Mac make a Dock Menu
+    if (process.platform === 'darwin') {
+      const {app, Menu} = require('electron');
```
Comment (rauchg) [inline app/index.js]: `standard` needs spaces when de-structuring IIRC. Did the `git push` linting not complain?

File: app/menu.js [hunk 4]
```diff
@@ -208,7 +208,7 @@ module.exports = function createMenu ({ createWindow, updatePlugins }) {
       ]
     },
     {
-      label: 'Window',
+      role: 'window',
```
Comment (rauchg) [inline app/menu.js]: nice!

**Comment by rauchg** [comment]
other than those nits, LGTM!

File: app/index.js (hunk 3 above)
Comment (MrRio) [inline app/index.js]: The linter didn't complain no, I think the rule it enforces it that you keep it consistent throughout that part.

**Pull request merged**